app/controllers/projects_controller.rb

```diff
@@ -110,11 +110,11 @@ def collaborators
 
   def follow
```

Comment:
And again. It maybe feels like creating a follower sits best in your authentication / authorisation logic.

Reply:
What did you have in mind? I'm struggling to see how I could extract much more from this, as the flash stuff belongs in the controller. The `current_user.follower_of?(@project)` bit is not governing permissions, but just setting up the endpoint as a toggle. (Which admittedly I hate, but I'm not changing existing behaviour for this pass)